### ISSUE TITLE
Pyinstaller FunAsr是否支持？能否给个详细的打包exe说明？

### ISSUE THREAD

**Issue by LinLinNa** [alibaba-damo-academy/FunASR#1726]
在工程中已安装好环境，并且可以运行（runtime\python\http\server.py）

使用Pyinstaller打包

    pyinstaller --hidden-import=modelscope --onefile .\server.py

打包成功，但运行出错，看上去运行了模型下载，但模型初始化似乎无效，报错结果如下：

![image](https://github.com/alibaba-damo-academy/FunASR/assets/48623316/b50d0ca0-4440-44b5-a181-39e9d11dc018)

环境及版本说明：

Package                   Version     Editable project location
------------------------- ----------- -------------------------
addict                    2.4.0
aiofiles                  23.2.1
aiohttp                   3.9.5
aiosignal                 1.3.1
aliyun-python-sdk-core    2.15.1
aliyun-python-sdk-kms     2.16.3
altgraph                  0.17.4
annotated-types           0.6.0
antlr4-python3-runtime    4.9.3
anyio                     4.3.0
async-timeout             4.0.3
attrs                     23.2.0
audioread                 3.0.1
beartype                  0.18.5
certifi                   2024.2.2
cffi                      1.16.0
charset-normalizer        3.3.2
click                     8.1.7
colorama                  0.4.6
crcmod                    1.7
cryptography              42.0.7
datasets                  2.18.0
decorator                 5.1.1
dill                      0.3.8
dnspython                 2.6.1
editdistance              0.8.1
einops                    0.8.0
email_validator           2.1.1
exceptiongroup            1.2.1
fastapi                   0.111.0
fastapi-cli               0.0.3
ffmpeg                    1.4
filelock                  3.14.0
frozenlist                1.4.1
fsspec                    2024.2.0
funasr                    1.0.26      d:\ai\funasr\new\funasr
gast                      0.5.4
h11                       0.14.0
httpcore                  1.0.5
httptools                 0.6.1
httpx                     0.27.0
huggingface-hub           0.23.0
hydra-core                1.3.2
idna                      3.7
importlib_metadata        7.1.0
intel-openmp              2021.4.0
jaconv                    0.3.4
jamo                      0.4.1
jieba                     0.42.1
Jinja2                    3.1.4
jmespath                  0.10.0
joblib                    1.4.2
kaldiio                   2.18.0
lazy_loader               0.4
librosa                   0.10.2
llvmlite                  0.42.0
markdown-it-py            3.0.0
MarkupSafe                2.1.5
mdurl                     0.1.2
mkl                       2021.4.0
modelscope                1.11.0
more-itertools            10.2.0
mpmath                    1.3.0
msgpack                   1.0.8
multidict                 6.0.5
multiprocess              0.70.16
networkx                  3.3
numba                     0.59.1
numpy                     1.26.4
omegaconf                 2.3.0
openai-whisper            20231117
orjson                    3.10.3
oss2                      2.18.5
packaging                 24.0
pandas                    2.2.2
pefile                    2023.2.7
pillow                    10.3.0
pip                       23.3.2
platformdirs              4.2.1
pooch                     1.8.1
protobuf                  5.26.1
pyarrow                   16.0.0
pyarrow-hotfix            0.6
pycparser                 2.22
pycryptodome              3.20.0
pydantic                  2.7.1
pydantic_core             2.18.2
Pygments                  2.18.0
pyinstaller               6.6.0
pyinstaller-hooks-contrib 2024.6
pynndescent               0.5.12
PyQt5                     5.15.10
PyQt5-Qt5                 5.15.2
PyQt5-sip                 12.13.0
python-dateutil           2.9.0.post0
python-dotenv             1.0.1
python-multipart          0.0.9
pytorch-wpe               0.0.1
pytz                      2024.1
pywin32-ctypes            0.2.2
PyYAML                    6.0.1
regex                     2024.5.10
requests                  2.31.0
rich                      13.7.1
rotary-embedding-torch    0.6.0
safetensors               0.4.3
scikit-learn              1.4.2
scipy                     1.13.0
sentencepiece             0.2.0
setuptools                69.0.3
shellingham               1.5.4
simplejson                3.19.2
six                       1.16.0
sniffio                   1.3.1
sortedcontainers          2.4.0
soundfile                 0.12.1
soxr                      0.3.7
starlette                 0.37.2
sympy                     1.12
tbb                       2021.12.0
tensorboardX              2.6.2.2
threadpoolctl             3.5.0
tiktoken                  0.6.0
tokenizers                0.19.1
tomli                     2.0.1
torch                     2.3.0+cu118
torch-complex             0.4.3
torchaudio                2.3.0
tqdm                      4.66.4
transformers              4.40.2
typer                     0.12.3
typing_extensions         4.11.0
tzdata                    2024.1
ujson                     5.9.0
umap-learn                0.5.6
urllib3                   2.2.1
uvicorn                   0.29.0
watchfiles                0.21.0
websockets                12.0
wheel                     0.42.0
xxhash                    3.4.1
yapf                      0.40.2
yarl                      1.9.4
zipp                      3.18.1

**Comment by LauraGPT** [comment]
https://github.com/alibaba-damo-academy/FunASR/discussions/1549

**Comment by wuzheng650** [comment]
使用Nuitka 打包的exe可以正常运行。

**Comment by Junyu-B** [comment]
Met exact same init problem, init issue is not covered in https://github.com/modelscope/FunASR/discussions/1549, any idea?

**Comment by yukozh** [comment]
+1

**Comment by Guolipeng666** [comment]
+1
